csvcubed/csvcubed/cli/inspect/inspectdatasetmanager.py

```diff
@@ -0,0 +1,298 @@
+    DatasetObservationsInfoResult,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+class DatasetMeasureType(Enum):
+    """
+    The type of dataset measure.
+    """
+
+    SINGLE_MEASURE = 0
+
+    MULTI_MEASURE = 1
+
+
+class DatasetUnitType(Enum):
+    """
+    The type of dataset unit.
+    """
+
+    SINGLE_UNIT = 0
+
+    MULTI_UNIT = 1
+
+
+def load_csv_to_dataframe(json_path: Path, csv_path: Path) -> DataFrame:
+    """
+    Loads the csv in given path to a Panda Dataframe.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `DataFrame` - Dataframe of the csv.
+    """
+
+    try:
+        dataset_path = json_path.parent / csv_path
+        _logger.debug(f"Dataset path: {dataset_path.absolute()}")
+
+        dataset = DataFrame(pd.read_csv(dataset_path))
+        _logger.info("Successfully loaded csv into dataframe.")
+        return dataset
+    except Exception as ex:
+        raise Exception("An error occured while loading csv into dataframe.") from ex
+
+
+def get_dataset_measure_type(
+    graph: Graph, dsd_uri: URIRef, json_path: Path
+) -> DatasetMeasureType:
+    """
+    Detects whether the dataset is single-measure or multi-measure.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `DatasetMeasureType`
+    """
+    result_qube_components: QubeComponentsResult = select_csvw_dsd_qube_components(
+        graph, dsd_uri, json_path
+    )
+    filtered_components = [
+        component
+        for component in result_qube_components.qube_components
+        if component.property_type == ComponentPropertyType.Measure.value
+    ]
+
+    measure_type = (
+        DatasetMeasureType.MULTI_MEASURE
+        if len(filtered_components) > 1
+        else DatasetMeasureType.SINGLE_MEASURE
+    )
+    _logger.debug(f"Dataset measure type: {measure_type}")
+
+    return (
+        DatasetMeasureType.MULTI_MEASURE
+        if len(filtered_components) > 1
+        else DatasetMeasureType.SINGLE_MEASURE
+    )
+
+
+def get_dataset_unit_type(graph: Graph, dsd_uri, json_path: Path) -> DatasetUnitType:
+    """
+    Detects whether the dataset is single-unit or multi-unit.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `DatasetUnitType`
+    """
+    result_qube_components: QubeComponentsResult = select_csvw_dsd_qube_components(
+        graph, dsd_uri, json_path
+    )
+    filtered_components = [
+        component
+        for component in result_qube_components.qube_components
+        if component.property == ComponentPropertyAttributeURI.UnitMeasure.value
+    ]
+
+    unit_type = (
+        DatasetUnitType.MULTI_UNIT
+        if len(filtered_components) > 0
+        else DatasetUnitType.SINGLE_UNIT
+    )
+    _logger.debug(f"Dataset unit type: {unit_type}")
+
+    return (
+        DatasetUnitType.MULTI_UNIT
+        if len(filtered_components) > 0
+        else DatasetUnitType.SINGLE_UNIT
+    )
+
+
+def get_measure_col_from_dsd(graph: Graph, dsd_uri, json_path: Path) -> str:
```

Comment:
@mikeAdamss Makes sense - dont know how I missed this! Should be because I was developing these incrementally and did not realise the previous function. I will get this sorted.